Find_first_missing_positive_integer.rb

```diff
@@ -0,0 +1,39 @@
+=begin 
+This problem was asked by Stripe.
+
+Given an array of integers, find the first missing positive integer in linear time and constant space. 
+In other words, find the lowest positive integer that does not exist in the array. 
+The array can contain duplicates and negative numbers as well.
+
+For example, the input [3, 4, -1, 1] should give 2. The input [1, 2, 0] should give 3.
+
+You can modify the input array in-place.
+
+=end 
+
+def missingpostiveint(array)
+    array = array.sort 
+    expected = array[0]
+    
+        
+ 
+
+    array.each_with_index do |value, index|
+        return expected  if value != expected
+        expected = 1  if expected == -1   
+        return  expected += 1  if  value == expected && index == array.length-1 
+        expected += 1         

```

Comment:
```suggestion
        return expected if value != expected
        expected = 1 if expected == -1
        return expected += 1  if value == expected && index == array.length-1
        expected += 1

```

So many spaces

Reply:
why skip 0 as a number? after -1 it should be 0, no?

Reply:
Thats what i had in my mind! but when it comes to this problem [3, 4, -1, 1] should give 2.
And when i  sort it the first smallest number should be 0 which is what i was getting, then when i saw the given answer i had to ignore 0 to get the expected answer.

Reply:
aha okey, i got you.